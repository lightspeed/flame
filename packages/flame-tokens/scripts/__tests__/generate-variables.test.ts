import variablesMap from '../generate-variables';

describe('generate-variables', () => {
  variablesMap.forEach(([fileName, output]) => {
    // eslint-disable-next-line jest/valid-describe
    describe(fileName, () => {
      it('should output the right CSS', () => {
        expect(output).toMatchSnapshot();
      });
    });
  });
});
