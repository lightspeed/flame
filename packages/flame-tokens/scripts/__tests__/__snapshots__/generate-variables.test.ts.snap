// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate-variables index.css should output the right CSS 1`] = `
":root {
  --cr-maple: #ed5153;
  --cr-maple-100: #fee3e3;
  --cr-maple-200: #b93435;
  --cr-maple-300: #872525;
  --cr-dive: #232a3b;
  --cr-dive-100: #3c7797;
  --cr-dive-200: #32637e;
  --cr-dive-300: #34495e;
  --cr-night: #0c0d0d;
  --cr-night-100: #383d3d;
  --cr-night-200: #212424;
  --cr-night-300: #181a1a;
  --cr-snow: #fff;
  --cr-snow-100: #fcfcfc;
  --cr-snow-200: #f3f3f3;
  --cr-snow-300: #e1e4e5;
  --cr-blue: #2875c6;
  --cr-blue-100: #e9fafe;
  --cr-blue-200: #5187e0;
  --cr-blue-300: #1d5794;
  --cr-green: #35b858;
  --cr-green-100: #e0f8e6;
  --cr-green-200: #3dd164;
  --cr-green-300: #26853f;
  --cr-orange: #e69524;
  --cr-orange-100: #fdebd0;
  --cr-orange-200: #f39c26;
  --cr-orange-300: #996318;
  --cr-yellow: #f1f326;
  --cr-yellow-100: #fdfcd4;
  --cr-yellow-200: #ecde36;
  --cr-yellow-300: #9e941f;
  --cr-gray: #494c4c;
  --cr-gray-100: #c4cacc;
  --cr-gray-200: #abb3b3;
  --cr-gray-300: #848a8a;
  --cr-seal: #737883;
  --cr-seal-100: #9ba0ab;
  --cr-seal-200: #505765;
  --cr-seal-300: #393f4c;
  --cr-serif: serif;
  --cr-sans-serif: Lato, Helvetica Neue, Helvetica, Arial, sans-serif;
  --cr-monospace: monospace;
  --cr-text-xxl: 2.25rem;
  --cr-text-xl: 1.5rem;
  --cr-text-l: 1.125rem;
  --cr-text: 1rem;
  --cr-text-s: 0.875rem;
  --cr-text-xs: 0.75rem;
  --cr-text-xxs: 0.5rem;
  --cr-regular: 400;
  --cr-bold: 700;
  --cr-letter-spacing-normal: normal;
  --cr-letter-spacing-1: 0.03125rem;
  --cr-letter-spacing-2: 0.0625rem;
  --cr-letter-spacing-3: 0.09375rem;
  --cr-spacing-0: 0;
  --cr-spacing-1: 0.375rem;
  --cr-spacing-2: 0.75rem;
  --cr-spacing-3: 1.125rem;
  --cr-spacing-4: 1.5rem;
  --cr-spacing-5: 1.875rem;
  --cr-spacing-6: 2.25rem;
  --cr-spacing-7: 2.625rem;
  --cr-spacing-8: 3rem;
  --cr-spacing-9: 3.375rem;
  --cr-spacing-10: 3.75rem;
  --cr-shadow-1: 0 0.0625rem 0.125rem rgba(12, 13, 13, 0.15);
  --cr-shadow-2: 0 0.125rem 0.25rem rgba(12, 13, 13, 0.15);
  --cr-shadow-3: 0 0.1875rem 0.375rem rgba(12, 13, 13, 0.15);
  --cr-shadow-4: 0 0.375rem 0.75rem rgba(12, 13, 13, 0.15);
  --cr-shadow-5: 0 0.75rem 1.5rem rgba(12, 13, 13, 0.15);
  --cr-inner-shadow-1: inset 0 0.0625rem 0.125rem rgba(56, 61, 61, 0.1);
  --cr-inner-shadow-2: inset 0 0 0.375rem rgba(12, 13, 13, 0.18);
  --cr-inner-shadow-n1: inset 0 -0.0625rem 0.1875rem rgba(12, 13, 13, 0.18);
  --cr-border-shadow: 0 0 0 0.0625rem rgba(12, 13, 13, 0.15);
  --cr-radius-1: 0.1875rem;
  --cr-radius-2: 0.375rem;
  --cr-radius-circle: 50%;
  --cr-transition-duration-fast: 100ms;
  --cr-transition-duration-base: 200ms;
  --cr-transition-duration-slow: 300ms;
}
"
`;

exports[`generate-variables index.scss should output the right CSS 1`] = `
"$cr-namespace: 'cr-';
$cr-maple: #ed5153;
$cr-maple-100: #fee3e3;
$cr-maple-200: #b93435;
$cr-maple-300: #872525;
$cr-dive: #232a3b;
$cr-dive-100: #3c7797;
$cr-dive-200: #32637e;
$cr-dive-300: #34495e;
$cr-night: #0c0d0d;
$cr-night-100: #383d3d;
$cr-night-200: #212424;
$cr-night-300: #181a1a;
$cr-snow: #fff;
$cr-snow-100: #fcfcfc;
$cr-snow-200: #f3f3f3;
$cr-snow-300: #e1e4e5;
$cr-blue: #2875c6;
$cr-blue-100: #e9fafe;
$cr-blue-200: #5187e0;
$cr-blue-300: #1d5794;
$cr-green: #35b858;
$cr-green-100: #e0f8e6;
$cr-green-200: #3dd164;
$cr-green-300: #26853f;
$cr-orange: #e69524;
$cr-orange-100: #fdebd0;
$cr-orange-200: #f39c26;
$cr-orange-300: #996318;
$cr-yellow: #f1f326;
$cr-yellow-100: #fdfcd4;
$cr-yellow-200: #ecde36;
$cr-yellow-300: #9e941f;
$cr-gray: #494c4c;
$cr-gray-100: #c4cacc;
$cr-gray-200: #abb3b3;
$cr-gray-300: #848a8a;
$cr-seal: #737883;
$cr-seal-100: #9ba0ab;
$cr-seal-200: #505765;
$cr-seal-300: #393f4c;
$cr-serif: serif;
$cr-sans-serif: Lato, Helvetica Neue, Helvetica, Arial, sans-serif;
$cr-monospace: monospace;
$cr-text-xxl: 2.25rem;
$cr-text-xl: 1.5rem;
$cr-text-l: 1.125rem;
$cr-text: 1rem;
$cr-text-s: 0.875rem;
$cr-text-xs: 0.75rem;
$cr-text-xxs: 0.5rem;
$cr-regular: 400;
$cr-bold: 700;
$cr-letter-spacing-normal: normal;
$cr-letter-spacing-1: 0.03125rem;
$cr-letter-spacing-2: 0.0625rem;
$cr-letter-spacing-3: 0.09375rem;
$cr-spacing-0: 0;
$cr-spacing-1: 0.375rem;
$cr-spacing-2: 0.75rem;
$cr-spacing-3: 1.125rem;
$cr-spacing-4: 1.5rem;
$cr-spacing-5: 1.875rem;
$cr-spacing-6: 2.25rem;
$cr-spacing-7: 2.625rem;
$cr-spacing-8: 3rem;
$cr-spacing-9: 3.375rem;
$cr-spacing-10: 3.75rem;
$cr-shadow-1: 0 0.0625rem 0.125rem rgba(12, 13, 13, 0.15);
$cr-shadow-2: 0 0.125rem 0.25rem rgba(12, 13, 13, 0.15);
$cr-shadow-3: 0 0.1875rem 0.375rem rgba(12, 13, 13, 0.15);
$cr-shadow-4: 0 0.375rem 0.75rem rgba(12, 13, 13, 0.15);
$cr-shadow-5: 0 0.75rem 1.5rem rgba(12, 13, 13, 0.15);
$cr-inner-shadow-1: inset 0 0.0625rem 0.125rem rgba(56, 61, 61, 0.1);
$cr-inner-shadow-2: inset 0 0 0.375rem rgba(12, 13, 13, 0.18);
$cr-inner-shadow-n1: inset 0 -0.0625rem 0.1875rem rgba(12, 13, 13, 0.18);
$cr-border-shadow: 0 0 0 0.0625rem rgba(12, 13, 13, 0.15);
$cr-radius-1: 0.1875rem;
$cr-radius-2: 0.375rem;
$cr-radius-circle: 50%;
$cr-transition-duration-fast: 100ms;
$cr-transition-duration-base: 200ms;
$cr-transition-duration-slow: 300ms;
"
`;
